### PR TITLE
feat(klaviyo): add all traits to profile properties in track call

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## Description of the change

- Add all traits to profile properties in track call

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/klaviyo/transform.js [hunk 1]
```diff
@@ -191,17 +187,16 @@ const trackRequestHandler = (message, category, destination) => {
...
   }
   // if flattenProperties is enabled from UI, flatten the event properties
   attributes.properties = flattenProperties
     ? flattenJson(attributes.properties, '.', 'normal', false)
     : attributes.properties;
   // Map user properties to profile object
-  attributes.profile = createCustomerProperties(message, destination.Config);
+  attributes.profile = {
+    ...createCustomerProperties(message, destination.Config),
+    ...flattenJson(populateCustomFieldsFromTraits(message)),
```
Comment (krishna2020) [inline src/v0/destinations/klaviyo/transform.js:198]: Having profile properties in flattened json by default is preferable ?

File: src/v0/destinations/klaviyo/transform.js
Comment (krishna2020) [inline src/v0/destinations/klaviyo/transform.js:1]: In Identify flow, does `data.attributes.properties` mean profile properties ?
https://github.com/rudderlabs/rudder-transformer/pull/2572/files#diff-7bf116831bc60affad855f69064fa73f991645698a1ea61b7e6f12badfc9fe1eR91

File: src/v0/destinations/klaviyo/transform.js (hunk 1 above)
Comment (krishna2020) [reply src/v0/destinations/klaviyo/transform.js:198]: This is not the case in identify flow

File: src/v0/destinations/klaviyo/transform.js
Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/transform.js:1]: Yes

File: src/v0/destinations/klaviyo/transform.js (hunk 1 above)
Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/transform.js:198]: Yes, this is preferable and we are doing it in `identify` as well in the line you pointed out in the next comment.

Comment (krishna2020) [reply src/v0/destinations/klaviyo/transform.js:199]: But it's driven via `flattenProperties` flag not by default

Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/transform.js]: Yeah, Misunderstood it. Updated it via `flattenProperties`.

**Comment by ujjwal-ab** [comment]
This is the PR to hotfix branch - https://github.com/rudderlabs/rudder-transformer/pull/2581

**Pull request closed without merging**